packages/react-admin-core/src/table/useTableQueryFilter.tsx

```diff
@@ -0,0 +1,19 @@
+import { AnyObject } from "final-form";
+import * as React from "react";
+
+export interface IFilterApi<FilterValues extends AnyObject> {
+    current: FilterValues;
+    changeFilters: (values: FilterValues) => void;
+}
+export function useTableQueryFilter<FilterValues extends AnyObject>(defaultValues: FilterValues): IFilterApi<FilterValues> {
+    const [filters, setFilters] = React.useState<FilterValues>(defaultValues);
```

Comment:
@nsams are filter-params now stored in two places? this state and tableFilterFinalForm-state?

Reply:
changing filter-params with changeFilters do not effect tableFilterFinalForm fields. do you think this could lead to confusion or misconception?

Reply:
> @nsams are filter-params now stored in two places? this state and tableFilterFinalForm-state?

yes, the one in useTableQueryFilter is debounced though

> changing filter-params with changeFilters do not effect tableFilterFinalForm fields. do you think this could lead to confusion or misconception?

Yes the API is in one direction only, and only a single filter form must be used. What are the alternatives?